boxoffice/models/line_item.py

```diff
@@ -114,32 +148,32 @@ def calculate_discounted_amount(self, discount_policy, line_item):
             return line_item.base_amount - discounted_price
         return (discount_policy.percentage * line_item.base_amount/Decimal(100))
 
+    def is_coupon_usable(self, coupon, applied_to_count):
+        return (coupon.usage_limit - LineItem.coupon_used_count(coupon)) > applied_to_count
```

Comment:
This is inefficient. You are now running a query on every GET request. What you need instead:

1. A `used_count` column on the DiscountCoupon model
2. An `update_used_count` method in DiscountCoupon that runs `self.used_count = select([func.count(LineItem.discount_coupon_id)]).where(LineItem.discount_coupon_id == coupon.id)` (do not execute the query!)
3. A call to this method every time LineItem is updated or edited.